### PR TITLE
usermodule-upload photo of user remaining, mail generating functionality remaing

### PR DESCRIPTION
**Pull request by niravparmar**
Adding comments 
- [x] Get All user
- [x] Get Specific User
- [x] login user
- [x] Signup User
- [x] Update User
- [x] Delete User
- [x] Verify User
- [x] forget password 
- [x] reset user password
- [x] Activate user password
- [x] change password encryption format sha256 to sha1

Reamaing:

- [ ] Upload photo of user
- [ ] adding mail send functionality : sending mail for reset-password after clicking forget-password
- [ ] dynamic token generation

User API's Postman Links
https://www.getpostman.com/collections/01b2d073ee85b173b234

### CODE REVIEW

File: src/actions/userActions.js [hunk 1]
```diff
@@ -1,71 +1,84 @@
...
+        })
+    },
 
-    UsersModel.update(id, req.body).then((userRes) => {
-      res.json({status: true, message:'Update SuccessFully!!!.... ', data:userRes})
-    })
+    updateUserById: function (req, res) {
+        const {id} = req.params;
+        UsersModel.update(id, req.body).then((userRes) => {
+            res.json({status: true, message: 'Update successfully', data: userRes})
```
Comment (suresh-jarvis) [inline src/actions/userActions.js]: What update successfully ?

File: src/model/users.js [hunk 2]
```diff
@@ -2,90 +2,104 @@ const mongoose = require('mongoose');
...
-      new Users(userData).save(function (err, data) {
-        resolve((err) ? false : data)
-      })
-    })
-  },
+    insert: function (userData) {
+        return new Promise((resolve, reject) => {
+            userData['createdAt'] = moment();
+            new Users(userData).save(function (err, data) {
+                if (err == null) {
```
Comment (suresh-jarvis) [inline src/model/users.js]: Use ternoery operator dont use if in single statement

Comment (suresh-jarvis) [reply src/model/users.js]: Dont use reject in query...

File: src/actions/userActions.js [hunk 3]
```diff
@@ -1,71 +1,98 @@
...
+            }
+        })
+    },
 
-    UsersModel.update(id, req.body).then((userRes) => {
-      res.json({status: true, message:'Update SuccessFully!!!.... ', data:userRes})
-    })
+    deleteUserById: function (req, res) {
+        const {id} = req.params;
+        UsersModel.update(id).then((userRes) => {
```
Comment (suresh-jarvis) [inline src/actions/userActions.js]: In Delete API you have used `UsersModel.update(id)` ? It should be `UsersModel.delete(id)`

File: src/model/users.js [hunk 4]
```diff
@@ -1,91 +1,111 @@
...
+    },
 
-  delete: function (id) {
-    return new Promise((resolve) => {
-      Users.findByIdAndRemove(id, function (error, data) {
-        resolve((error) ? {} : (data == null) ? {} : data)
-      })
-    })
-  },
+    update: function (id, data = 0) {
```
Comment (suresh-jarvis) [inline src/model/users.js]: You can not use same method for Update and Delete user ? It's not developer friendly

File: src/actions/userActions.js [hunk 5]
```diff
@@ -1,71 +1,98 @@
...
+                if (Object.keys(data).length > 0) {
+                    res.json({status: false, message: 'UserName is already exists'});
+                } else {
+                    const {firstName, lastName, userName, email, password} = req.body;
+                    const userData = {
+                        firstName,
+                        lastName,
+                        userName,
+                        email,
+                        password: crypto.createHmac('sha256', cypherKey)
```
[the remark this answers is not in the document]
Comment (niravparmar) [reply src/actions/userActions.js]: it is outdated file, i'll already updated it

**Pull request merged**